enclave/core/sgx/keys.c

```diff
@@ -218,6 +220,38 @@ static oe_result_t _get_default_key_request_attributes(
     return result;
 }
 
+static oe_result_t _get_seal_key_info_internal(
+    oe_seal_policy_t seal_policy,
+    uint64_t opt_flags,
+    sgx_key_request_t *key_request)
+{
+    oe_result_t result = OE_OK;
+
+    OE_CHECK(_get_default_key_request_attributes(key_request));
+
+    // Set custom attributes mask if provided
+    if (opt_flags & OE_SEAL_SGX_SPECIFIC)
+        key_request->attribute_mask.flags = opt_flags & ~OE_SEAL_SGX_SPECIFIC;
```

Comment:
This really shouldn't combine an indicator for the TEE type with flags that have a specific meaning in a TEE (where the bit that the TEE type takes up has overloaded meaning now). It also makes it impossible to extend for other TEEs because you can't actually take the `opt_flags` and check other bits to figure out which TEE they should be interpreted as.

It seems like `OE_SEAL_SGX` should be one of potentially many enum values, and the opt_flags themselves should be split out into some opaque TEE-specific data structure in a manner similar to the `oe_enclave_settings_t`

Reply:
`OE_SEAL_SGX` is defined **if and only if** SGX is the target TEE. The expectation is that, for each TEE supported by OE, there'll be an `OE_SEAL_<TEE>` macro defined **if and only if** that TEE is the target TEE. Therefore, the expression `OE_SEAL_TEE1 | OE_SEAL_TEE2` can never be valid and will always result in compiler errors.

That said, `OE_SEAL_SGX` is **not** a TEE indicator. `oe_get_seal_key_info()` is always TEE specific (as `key_info` is TEE specific) so has no need to test TEE type. In fact, on SGX, `oe_get_seal_key_info()` can always treat `opt_flags` as `attribute_mask` without testing anything. The intention of defining `OE_SEAL_SGX` to a non-zero value is to force it in SGX specific code, so as to improve readability and to err on other TEEs.